app/src/main/java/com/example/diceroller/RolledDiceViewModel.kt

```diff
@@ -1,65 +0,0 @@
-package com.example.diceroller
-
-import androidx.lifecycle.ViewModel
-import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.asStateFlow
-import java.text.DateFormat
-import kotlin.random.Random
-
-class RolledDiceViewModel : ViewModel() {
-    private val _currentDice: MutableStateFlow<DiceType> = MutableStateFlow(DiceType.D6)
-    val currentDice = _currentDice.asStateFlow()
-
-    private val _listOfDices: MutableStateFlow<List<Dice>> = MutableStateFlow(listOf<Dice>())
-    val listOfDices = _listOfDices.asStateFlow()
-
-    fun getHistoryOfDices(): List<Dice> = listOfDices.value
-
-    fun addDiceToList(dice: Dice) {
-        _listOfDices.value = listOfDices.value+dice
-    }
-
-    //    Deprecated, changed to triggers
-    fun switchCurrentDiceType(diceType: DiceType) {
-        _currentDice.value = diceType
-    }
-
-    fun getCurrentDiceType(): DiceType = currentDice.value
-
-
-
-    fun rollDice():Dice {
-        val diceRange = getCurrentDiceType().diceRange
-        val diceResult = diceRange.random(Random(System.currentTimeMillis()))
-
-        val newDice = Dice(getCurrentDiceType(), diceResult, java.util.Calendar.getInstance())
-        addDiceToList(newDice)
-
-        return newDice
-    }
-
-
-    fun triggerD4() {
-        _currentDice.value = DiceType.D4
-    }
-
-    fun triggerD6() {
-        _currentDice.value = DiceType.D6
-    }
-
-    fun triggerD8() {
-        _currentDice.value = DiceType.D8
-    }
-
-    fun triggerD10() {
-        _currentDice.value = DiceType.D10
-    }
-
-    fun triggerD12() {
-        _currentDice.value = DiceType.D12
-    }
-
-    fun triggerD20() {
-        _currentDice.value = DiceType.D20
-    }
```

Comment:
👍  So close to something so clean. This is definitely what I mean by telling the ViewModel the user did something. But this is so repetitive. Keep your code DRY. DRY = Don't Repeat Yourself